### DISCUSSION TITLE
Pattern matching for restricting workflows to run in a runner group

### DISCUSSION THREAD

**Discussion by adrielldagasuan** [actions/runner#3531] [Ideas]
https://docs.github.com/en/enterprise-cloud@latest/actions/hosting-your-own-runners/managing-self-hosted-runners/managing-access-to-self-hosted-runners-using-groups#changing-which-workflows-can-access-an-organization-runner-group

One use case is that we're using reusable workflows to run in a specific runner group (with extra permissions) and nothing else can run in that. However, whenever we release a new version of that workflow, we have to add that specific ref to the whitelist and ultimately that list will just grow.

We do already have the capability of whitelisting actions with wildcards as reference. https://docs.github.com/en/organizations/managing-organization-settings/disabling-or-limiting-github-actions-for-your-organization so why not apply the same rules to the allowed workflows.

**Comment by tcowen-d2l** [comment]
Is it possible to bump this? Our only option is to not restrict our runner groups to any workflows because of the ref pinning requirement. The _only_ time ref pinning would be useful is for example when runners should only ever run against the `main` branch, but since a lot of runners are needed for PR branches as well, pinning to specific refs is impossible.

**Comment by adrielldagasuan** [comment]
Bumping this up as this is something that would greatly improve management of workflows to runner groups. At the moment, we're pinning everything to the default branch and it's very hard to introduce breaking changes with this setup.

**Comment by alileza** [comment]
can we have this please? seems like a tiny feature
